src/tests/JIT/Regression/JitBlue/Runtime_91175/Runtime_91175.cs

```diff
@@ -0,0 +1,19 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.aa
+
+using System.Runtime.CompilerServices;
+using System.Runtime.Intrinsics;
+using System.Runtime.Intrinsics.X86;
+using Xunit;
+
+public class TestClass
+{
+    [MethodImpl(MethodImplOptions.NoInlining)]
+    public static Vector256<int> Method0() => Avx2.ShiftRightArithmetic(Vector256<int>.AllBitsSet, Vector128<int>.AllBitsSet);
+
+    [Fact]
+    public static void TestEntryPoint()
+    {
+        _ = Method0();
```

Comment:
The test is sufficient to cover the reported issue. However, it would be nice that the test checks numerical correctness as well.

Reply:
We have a number of other tests covering numerical correctness of the functionality.

This is namely a regression test covering a very specific assert.

Reply:
Had there was a numerical correctness test case covering this specific case, it would had failed on CI before this fix, right? I don't see this PR enabling any existing test. Thus, I suspect that this is sufficient numerical correctness test coverage for the scenario that this PR is fixing.

Reply:
We have tests validating it here: https://github.com/dotnet/runtime/blob/main/src/tests/Common/GenerateHWIntrinsicTests/GenerateHWIntrinsicTests_X86.cs#L347C1-L348

This is an edge case around value numbering in a very particular scenario and ensuring we don't assert. We were already effectively doing the right thing codegen wise.

Reply:
I assume those tests aren't covering the VN constant folding, or we would have found this issue and #88451 as well. Can we somehow validate basic correctness of the rest of the constant folding that was added in .NET 8?

> This pr is fixing an edge case around value numbering in a very particular scenario and ensuring we don't assert. We were already effectively doing the right thing codegen wise.

The release behavior here is to hit a noway assert, so I think we should backport this fix to .NET 8.

Reply:
Let's consider adding separate testing separately from this so we can get going on the backport.